### DISCUSSION TITLE
Final CraftItemEvent result

### DISCUSSION THREAD

**Discussion by altruiis** [PaperMC/Paper#10552] [Ideas]
### Is your feature request related to a problem?

The `CraftItemEvent` does not allow you to see how many items the player actually crafted. It seems like the result acts more like a copy. It would be a lot more intuitive if the event offered the number of items the player *actually* crafted.

### Describe the solution you'd like.

Some way to access the final result of the craft event, like getFinalResult or just altering the current method that already exists to make it reflect the result of the event.

### Describe alternatives you've considered.

Hacky & imperfect guessing logic to see how many items the player crafted based on their inventory.

### Other

The original issue explains this pretty well: #7383. I am opening this issue because that one was closed due to being stale.
Also, IIRC the `PlayerTradeEvent` offers a proper result that tells you how many emeralds or other items the player received, and works when the item is shift-clicked into the player's inventory. Maybe the `CraftItemEvent` could be updated to look more like that, if that's possible? Thank you for reading!

**Comment by Doc94** [comment]
Hmm i check a little the logic and looks like this.

1. Player put items 3 stacks of wheet
2. Player use Shift+Click to craft all
3. Server craft the first item (1 bread)
4. Server process all the rest to try to move into inventory (63 bread)

If only need call the NMS method for calculate the mount of items maybe can make a PR for this and wait a few tests...

**Comment by Machine-Maker** [comment]
Related to https://github.com/PaperMC/Paper/issues/8749
